pkg/module/utils.go

```diff
@@ -843,3 +844,51 @@ func getHashBasedRepoPath(repoURL, hash string) (string, error) {
 
 	return filepath.Join(REPO_DIR, host, owner, repo, hash), nil
 }
+
+// File inputs
+
+func HasInputFiles(inputFiles data.InputFiles) bool {
+	return len(inputFiles.Required) > 0 || len(inputFiles.Optional) > 0
+}
+
+// Check that required files exist and only explicitly defined
+// files are in the target directory
+func ValidateInputFiles(path string, inputFiles data.InputFiles) error {
```

Comment:
What do you think about having an additional check inside this function to prevent like .exe and other malicious things from coming in?

We could come up with a list of allowed types or a list of disallowed types depending on which list will be shorter

Reply:
We could, but I'm not sure how much it improves our security.

Consider this scenario. A module author specifies a required `mybad.txt` file, then once the file is available to the module code, they rename it to `mybad.exe`.

Reply:
Ah gotcha, yeah, I wonder if its just a matter of adding a check in the RP side down the line to have them have a file check or something to filter out file formats to exclude from processing

Reply:
Yeah, we could do some analysis on the files. Go deeper than just validating file names. Do you have ideas for how we would approach this?

Reply:
Hmm not off the top of my head but its something we can brainstorm on together, i'm sure there's a way

Reply:
Added some basic checks here: https://github.com/Lilypad-Tech/lilypad/pull/573/commits/1b074d1f8eea20a3ce21dbd8ada315676f2c94a1